python/cudf/cudf/core/column/column.py

```diff
@@ -1364,9 +1371,18 @@ def as_column(arbitrary, nan_as_null=None, dtype=None, length=None):
             data = as_column(pa.array(arbitrary, from_pandas=True))
         elif arbitrary.dtype == np.bool:
             # Bug in PyArrow or HDF that requires us to do this
+            data = as_column(np.asarray(arbitrary), dtype=arbitrary.dtype,)
```

Comment:
Any reason we shouldn't just go straight through cupy here similarly to the other branches?
```suggestion
            data = as_column(cupy.asarray(arbitrary), dtype=arbitrary.dtype,)
```

Reply:
@kkraus14, yep don't see a reason why we shouldn't use `cupy`. Checked the tests locally LGTM.